x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -155,6 +164,17 @@ static AggValueExtractor getExtractor(Aggregation aggregation) {
             return TYPE_VALUE_EXTRACTOR_MAP.get(SingleBucketAggregation.class.getName());
         } else if (aggregation instanceof MultiBucketsAggregation) {
             return TYPE_VALUE_EXTRACTOR_MAP.get(MultiBucketsAggregation.class.getName());
+        } else if (aggregation instanceof InternalAggregation) {
```

Comment:
This is frustrating. If I could move the `InternalGeoLine` aggregation class over to xpack core, I think we can clean this up and it would even be a performance improvement as we won't need to serialize to xcontent and extract from the map.

@talevy think we can move `InternalGeoLine` to xpack core?

Reply:
Well, another solution would be to add a new interface that is contained in spatial core that `InternalGeoLine` implements. That isn't too bad IMO. We just need a way to grab the geo_shape formatted information from within the `InternalGeoLine`.